app/workers/preparsing.rb

```diff
@@ -44,36 +44,36 @@ def perform(genotype_id)
 
     if genotype.filetype == "23andme"
       # first non-comment line is of length 4 after split
-      if l.split("\t").length == 4
+      if l.strip.split("\t").length == 4
         logger.info "file is 23andme and is ok!"
         file_is_ok = true
       end
     elsif genotype.filetype == "ancestry"
       # first line is of length 5
-      if l.split("\t").length == 5
+      if l.strip.split("\t").length == 5
         file_is_ok = true
         logger.info "file is ancestry and is ok!"
       end
     elsif genotype.filetype == "decodeme"
       # first line is of length 6
-      if l.split(",").length == 6
+      if l.strip.split(",").length == 6
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.